test/safe/push.test.ts

```diff
@@ -3,14 +3,65 @@ import nock from "nock";
 import { createJobData, processPushJob } from "../../src/transforms/push";
 import { createWebhookApp } from "../utils/probot";
 import {getLogger} from "../../src/config/logger";
+import {mocked} from "ts-jest/utils";
+import {Installation, Subscription} from "../../src/models";
+import {booleanFlag, BooleanFlags} from "../../src/config/feature-flags";
+import {when} from "jest-when";
+import waitUntil from "../utils/waitUntil";
+import {start, stop} from "../../src/worker"
 
-describe("GitHub Actions", () => {
+jest.mock("../../src/models");
+jest.mock("../../src/config/feature-flags");
+
+describe("GitHub Push", () => {
 	let app;
-	beforeEach(async () => app = await createWebhookApp());
+	const mockGithubAccessToken = () => {
+		githubNock
+			.post("/app/installations/1234/access_tokens")
+			.reply(200, {
+				token: "token",
+				expires_at: new Date().getTime()
+			});
+	}
+
+	beforeEach(async () => {
+
+		mocked(Subscription.getAllForInstallation).mockResolvedValue([
+			{
+				jiraHost: process.env.ATLASSIAN_URL,
+				gitHubInstallationId: 1234,
+				enabled: true
+			}] as any);
+		mocked(Subscription.getSingleInstallation).mockResolvedValue(
+			{
+				id: 1,
+				jiraHost: process.env.ATLASSIAN_URL
+			} as any);
+		mocked(Installation.getForHost).mockResolvedValue(
+			{
+				jiraHost: process.env.ATLASSIAN_URL,
+				sharedSecret: process.env.ATLASSIAN_SECRET,
+				enabled: true
+			} as any
+		);
+
+		app = await createWebhookApp();
+	});
+
+	afterEach(() => {
+
+		if (!nock.isDone()) {
+			// eslint-disable-next-line jest/no-jasmine-globals
+			fail("nock is not done yet");
+		}
+		nock.cleanAll();
+		jest.restoreAllMocks();
+	});
 
 	const getAtlassianUrl = () => process.env.ATLASSIAN_URL || "";
 	const createJob = (payload, jiraHost:string) => ({data: createJobData(payload, jiraHost)} as any);
 
+	//TODO
 	describe.skip("add to push queue", () => {
```

Comment:
The tests which test that the message was send to the queue are not there. 

However there are some queue-agnostic end-to-end test below